src/server/scripts/World/npcs_special.cpp

```diff
@@ -299,101 +299,79 @@ class npc_chicken_cluck : public CreatureScript
 
 enum DancingFlames
 {
-    SPELL_BRAZIER           = 45423,
-    SPELL_SEDUCTION         = 47057,
-    SPELL_FIERY_AURA        = 45427
+    SPELL_SUMMON_BRAZIER    = 45423,
+    SPELL_BRAZIER_DANCE     = 45427,
+    SPELL_FIERY_SEDUCTION   = 47057
 };
 
-class npc_dancing_flames : public CreatureScript
+struct npc_dancing_flames : public ScriptedAI
 {
-public:
-    npc_dancing_flames() : CreatureScript("npc_dancing_flames") { }
+    npc_dancing_flames(Creature* creature) : ScriptedAI(creature) { }
 
-    struct npc_dancing_flamesAI : public ScriptedAI
+    void Reset() override
     {
-        npc_dancing_flamesAI(Creature* creature) : ScriptedAI(creature)
-        {
-            Initialize();
-        }
-
-        void Initialize()
-        {
-            Active = true;
-            CanIteract = 3500;
-        }
-
-        bool Active;
-        uint32 CanIteract;
+        DoCastSelf(SPELL_SUMMON_BRAZIER, true);
+        DoCastSelf(SPELL_BRAZIER_DANCE, false);
+        me->SetUInt32Value(UNIT_NPC_EMOTESTATE, EMOTE_STATE_DANCE);
+        float x, y, z;
+        me->GetPosition(x, y, z);
+        me->Relocate(x, y, z + 1.05f);
```

Comment:
try updateposition

Reply:
```cpp
        float x, y, z, o;
        me->GetPosition(x, y, z, o);
        me->UpdatePosition(x, y, z + 1.05f, o);
```
Works same way, but why it's used only in ancient scripts and new ones uses Relocate?

Reply:
updateposition is the legit way of changing a poisition, since it goes through all the "update position" related logic

relocate is a straight serverside position change. position is later on sent to client through other means

though maybe a specific situation does not require the full "update position" logic, relocate should be used knowing the somewhat bypass that it implies.

Reply:
Then Relocate sounds better here